crates/common/mqtt_channel/src/session.rs

```diff
@@ -0,0 +1,87 @@
+use crate::{Config, Connection, MqttError};
+use rumqttc::{AsyncClient, Event, Packet};
+
+/// Create a session using the `config.session_name`
+///
+/// The config can be used to connect later using `Connection::new(config)`.
+/// All the messages that have been published meantime with `QoS > 1`
+/// on the `config.subscriptions` topics will be received by the new connection.
+///
+/// `mqtt_channel::init_session(&config) consumes no messages.
+pub async fn init_session(config: &Config) -> Result<(), MqttError> {
```

Comment:
What is the purpose of this function? It doesn't return a client or anything and takes an immutable borrow. To me that suggests that it either does not do much, or it modifies _global_ state. The latter of which is not clearly documented nor any possible shortcomings of that method.


Looking at the documentation of `AsyncClient`/`EventLoop` I also don't see any `Drop` implementations, meaning that there is also the potential of leaving a connection 'open' and letting the server let it time out.

Reply:
Additionally: What happens if this is called multiple times:

- With the same argument
- With a different argument?

Reply:
> To me that suggests that it either does not do much, or it modifies global state.

It modifies a global state in the MQTT server pointed to by the `config`, creating a session with the given name and set of subscriptions. This function can be called multiple times with the same session name.

=> I will add this to the doc comment.

> Looking at the documentation of AsyncClient/EventLoop I also don't see any Drop implementations, meaning that there is also the potential of leaving a connection 'open' and letting the server let it time out.

This is a good point. => I will add an explicit `client.disconnect()`.